src/Simplex/Chat.hs

```diff
@@ -909,6 +822,79 @@ processChatCommand' vr = \case
+            pure (ct, cci)
+          mc <- forwardMC ci
+          file <- forwardCryptoFile ci `catchChatError` \_ -> pure Nothing
+          let ciff = forwardCIFF ci $ CIFFContact (forwardName ct) fromChatId
+          pure (ComposedMessage file Nothing mc, ciff)
+          where
+            forwardName :: Contact -> ContactName
+            forwardName Contact {profile = LocalProfile {displayName, localAlias}}
+              | localAlias /= "" = localAlias
+              | otherwise = displayName
+        CTGroup -> do
+          (gInfo, CChatItem _ ci) <- withStore $ \db -> do
+            gInfo <- getGroupInfo db vr user fromChatId
+            cci <- getGroupChatItem db user fromChatId itemId
+            pure (gInfo, cci)
+          mc <- forwardMC ci
+          file <- forwardCryptoFile ci `catchChatError` \_ -> pure Nothing
+          let ciff = forwardCIFF ci $ CIFFGroup (forwardName gInfo) fromChatId
+          pure (ComposedMessage file Nothing mc, ciff)
+          where
+            forwardName :: GroupInfo -> ContactName
+            forwardName GroupInfo {groupProfile = GroupProfile {displayName}} = displayName
+        CTLocal -> do
+          (CChatItem _ ci) <- withStore $ \db -> getLocalChatItem db user fromChatId itemId
+          mc <- forwardMC ci
+          file <- forwardCryptoFile ci `catchChatError` \_ -> pure Nothing
+          let ciff = forwardCIFF ci $ CIFFNoteFolder "notes" fromChatId
+          pure (ComposedMessage file Nothing mc, ciff)
+        CTContactRequest -> throwChatError $ CECommandError "not supported"
+        CTContactConnection -> throwChatError $ CECommandError "not supported"
+        where
+          forwardMC :: ChatItem c d -> CM MsgContent
+          forwardMC ChatItem {meta = CIMeta {itemDeleted = Just _}} = throwChatError CEInvalidForward
+          forwardMC ChatItem {content = CISndMsgContent fmc} = pure fmc
+          forwardMC ChatItem {content = CIRcvMsgContent fmc} = pure fmc
+          forwardMC _ = throwChatError CEInvalidForward
+          forwardCIFF :: ChatItem c d -> CIForwardedFrom -> CIForwardedFrom
+          forwardCIFF ChatItem {meta = CIMeta {itemForwarded = Just ciff}} _ = ciff
+          forwardCIFF _ ciff = ciff
+          forwardCryptoFile :: ChatItem c d -> CM (Maybe CryptoFile)
+          forwardCryptoFile ChatItem {file = Just CIFile {fileName, fileSource = Just cf@CryptoFile {filePath}}} =
+            chatReadVar filesFolder >>= \case
+              Nothing ->
+                ifM (doesFileExist filePath) (pure $ Just cf) (pure Nothing)
+              Just filesFolder ->
+                ifM
+                  (doesFileExist filePath)
+                  ( do
+                      fPath <- liftIO $ filesFolder `uniqueCombine` fileName
+                      liftIO $ copyFile filePath fPath -- to keep forwarded file in case original is deleted
+                      pure $ Just (cf {filePath = fPath} :: CryptoFile)
```

Comment:
it's not "key re-use" since content is the same, we're just copying file and metadata